std/internal/math/biguintcore.d

```diff
@@ -719,7 +719,7 @@ public:
         uint words = cast(uint)(y >> LG2BIGDIGITBITS);
         if (bits == 0)
         {
-            return BigUint(data[words..$]);
+            return BigUint(data[words..$].idup); // TODO can .idup be avoided?
```

Comment:
Is there a way to avoid this error message with the `.idup` here? I don't understand why `data[words..$]` can't be passed to `BigUint.ctor()`as data is an array of immutable values. Is it because this in this `opBinary` is qualified as scope?

Reply:
Making opBinary non-`scope` did it.

Reply:
`immutable` is irrelevant, you annotated the function `scope` so `this.data` is `scope` and can't be passed to a non-scope constructor.